pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -665,8 +665,9 @@ func (s *DiscoveryServer) startPush(push *model.PushContext, full bool, edsUpdat
 		adsLog.Infof("Starting new push while %v were still pending", currentlyPending)
 	}
 	startTime := time.Now()
+	pushEvent := &PushEvent{edsUpdates, push, startTime, full}
 	for _, p := range pending {
-		s.pushQueue.Enqueue(p, &PushInformation{edsUpdates, push, startTime, full})
+		s.pushQueue.Enqueue(p, pushEvent)
```

Comment:
This is broken, each event push event needs to be distinct or you will have race conditions. Please keep it in the loop

Reply:
Let me have a look

Reply:
I think i know what you mean, it may happen when the queue accumulated